packages/template-retail-react-app/app/components/island/README.md

```diff
@@ -0,0 +1,50 @@
+# 🏝️️Islands
+
+Inspired by [Astro’s islands architecture](https://docs.astro.build/en/concepts/islands/), this component is intended to give developers explicit and fine-granular control over the hydration behavior of their experiences.
+
+> ℹ️ Hydration is the process of making a server-rendered HTML page interactive, i.e. bridging the gap between server-side rendering (SSR) and client-side interactivity.
+
+# 😶‍🌫️ Hydration’s Curse
+
+Our Composable Storefront applications are server-rendered Single-Page Applications (SPAs). During hydration, the client takes over the server-rendered HTML, restores client-side interactivity and transitions the application into its fully-functional SPA mode.
+
+Behind the scenes, this **restoration of interactivity is computationally very expensive**. This may sound surprising, since hydration from the outside seems to only be about applying certain event listeners to certain elements. Unfortunately, the reality is much more complex. Even though the HTML/DOM already exists, **React must reconcile and often re-execute render logic across the entire application**, which can feel like a re-render and be similarly costly.
```

Comment:
@seckardt if this is a "manifesto" as it appears, I might specifically point out that the React "DOM tree thrashing" effect is a powerful negative player in the dynamic. Each change as hydration lands can introduce stateful behaviors that are difficult / impossible to reason about, as they send changes to the top of the in-memory React state tree. Those changes ripple down the tree and they do so as each potentially-stateful change lands. The end result is something that looks performant on a brand new Macbook, but sucks on an outdated mobile device

Reply:
That's a whole separate story even. DOM tree thrashing, forced synchronous layouts, or layout thrashing are topics we need to get to as well. But they're unrelated to the typical TBT-impacting rehydration problems of framworks like React, LWC, etc. So let's keep things separated.